plugins/search/src/components/SearchTypeFacet/SearchTypeFacet.tsx

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  ChangeEvent,
+  cloneElement,
+  Fragment,
+  useEffect,
+  useState,
+} from 'react';
+import { useSearch } from '../SearchContext';
+import {
+  Accordion,
+  AccordionSummary,
+  AccordionDetails,
+  Card,
+  CardContent,
+  CardHeader,
+  Divider,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  makeStyles,
+} from '@material-ui/core';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
+import AllIcon from '@material-ui/icons/FontDownload';
+
+const useStyles = makeStyles(theme => ({
+  card: {
+    backgroundColor: 'rgba(0, 0, 0, .11)',
+  },
+  cardContent: {
+    paddingTop: theme.spacing(1),
+  },
+  icon: {
+    color: theme.palette.common.black,
+  },
+  list: {
+    width: '100%',
+  },
+  listItemIcon: {
+    width: '24px',
+    height: '24px',
+  },
+  accordion: {
+    backgroundColor: theme.palette.background.paper,
+  },
+  accordionSummary: {
+    minHeight: 'auto',
+    '&.Mui-expanded': {
+      minHeight: 'auto',
+    },
+  },
+  accordionSummaryContent: {
+    margin: theme.spacing(2, 0),
+    '&.Mui-expanded': {
+      margin: theme.spacing(2, 0),
+    },
+  },
+  accordionDetails: {
+    padding: theme.spacing(0, 0, 1),
+  },
+}));
+
+/**
+ * @public
+ */
+export type SearchTypeFacetProps = {
+  /* what about this? */
```

Comment:
> /* what about this? */

? 😬

Reply:
Heh.  Was trying to document the types here (in such a way that they'd be detected and shown to someone using the component), but apparently this is not a thing.

https://github.com/TypeStrong/typedoc/issues/278#issuecomment-671224506